028-service-binding.md

```diff
@@ -0,0 +1,340 @@
+
+The Service Binding specification defines how connection information (e.g. endpoints, username, password) for services such as databases and message brokers are made available to runtime applications in Kubernetes. Version 1 of the specification is due to be released shortly.
+
+Today, Strimzi does not fit very nicely with service binding, it both lacks the status field required by the Service Binding operator and does not make credentials available in a single Secret which is what the Service Binding Operator expects from services that implement the spec.
+
+Contents:
+
+ - [Current problems with integrating Strimzi and a Service Binding Operator](#current-problems-with-integrating-strimzi-and-a-service-binding-operator)
+ - [Proposal](#proposal)
+ - [Overview of Service Binding specification](#overview-of-service-binding-specification)
+ - [Connecting to Strimzi](#connecting-to-strimzi)
+
+## Current problems with integrating Strimzi and a Service Binding Operator
+
+At a high level these are the current problems that make integrating Strimzi with a Service Binding Operator hard:
+
+1. Bootstrap Server information is not contained in any secret
+2. Information is spread across multiple `Secret` resources
+3. The user has to determine which `KafkaUser` can access which listener
+
+## Proposal
+
+I do not see a single approach that easily solves this problem. I have detailed a few below to enable the Strimzi and Service Binding specification communities to discuss how these two technologies should best be integrated.
+
+Proposed approaches:
+1. Application uses `KafkaUser` to get credentials but does not use the service binding mechanism for cluster certificate or bootstrap server
+2. Application uses multiple `ServiceBinding` resources to get information
+3. Application binds to the `KafkaUser` resource which contains all required information
+4. Application can bind to a single custom resource to get all information
+
+### 1 - Application binds to KafkaUser only
+
+In this approach the application creates a `ServiceBinding` for a specific `KafkaUser` CR. The credential information is bound to the application and the cluster certificate and bootstrap server address is obtained by the application in another way.
+
+Required changes:
+
+ - Update the `KafkaUser` CR to include a `status.binding` field where the `name` matches the current field `secret` in the status
+ - Update the secret created for the `KafkaUser` to include the username when the `KafkaUser` is using type SASL SCRAM
```

Comment:
This should not be a problem. I assume that means that the service binding will pass username / password and not the JAAS configuration? That would make it suitable with non-java clients. But make it harder to use with Java clients I guess.

Reply:
I think the service binding spec allow the secret to contain other fields. It just requires that field with the names defined in the spec have the value/meaning defined in the spec.

Reply:
Yes a service binding operator will mount all fields in the secret, so Java clients could use the JAAS configuration and non-Java clients can use the username/password